python/packages/ai/tests/auth/test_sso_auth.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+from unittest.mock import MagicMock, patch, AsyncMock
+from datetime import datetime, timedelta
+from botbuilder.core import TurnContext
+from msal import ConfidentialClientApplication
+from botbuilder.schema import Activity, ActivityTypes, ConversationAccount, ChannelAccount
+from botbuilder.schema.teams import TeamsChannelAccount
+
+from teams.state import TurnState
+from teams.auth import SsoOptions, ConfidentialClientApplicationOptions
+from teams.auth import SsoAuth
+
+class TestSsoAuthTokenExchange(unittest.IsolatedAsyncioTestCase):
```

Comment:
title of test usually matches name of file (e.g., `TestSsoAuth`) to allow for easier tracing